lib/wordnet/synset.rb

```diff
@@ -100,13 +198,18 @@ def hypernym
       relation(HYPERNYM)[0]
     end
 
+    # Get the parent synset (higher-level category, i.e. fruit -> reproductive_structure).
+    def hypernyms
+      relation(HYPERNYM)
+    end
+
     # Get the child synset(s) (i.e., lower-level categories, i.e. fruit -> edible_fruit)
-    def hyponym
+    def hyponyms
       relation(HYPONYM)
     end
 
     # Get the entire hypernym tree (from this synset all the way up to +entity+) as an array.
-    def expanded_hypernym
+    def expanded_first_hypernyms
```

Comment:
so the old hypernym function only returns the first hypernym, and i added a hypernyms function that returns all hypernyms of a synset. I also changed the `expanded_hypernyms` function so that it includes all hypernyms at every level. This is the old implementation of `expanded_hypernyms` that only use the first hypernym at every level.